pkg/crypto/tinkcrypto/primitive/aead/subtle/aes_cbc.go

```diff
@@ -127,34 +112,46 @@ func newCipher(key, iv []byte, decrypt bool) (cipher.BlockMode, error) {
 		return nil, fmt.Errorf("aes_cbc: failed to create block cipher, error: %w", err)
 	}
 
-	// If the IV is less than BlockSize bytes we need to pad it with zeros otherwise NewCBCEncrypter will panic.
+	// IV size must be at least BlockSize.
 	if len(iv) < aes.BlockSize {
-		paddedIV := make([]byte, aes.BlockSize)
-		if n := copy(paddedIV, iv); n != len(iv) {
-			return nil, errors.New("aes_cbc: failed to pad IV")
-		}
-
-		if !decrypt {
-			return cipher.NewCBCEncrypter(block, paddedIV), nil
-		}
-
-		return cipher.NewCBCDecrypter(block, paddedIV), nil
+		return nil, errors.New("aes_cbc: invalid iv size")
 	}
 
-	if !decrypt {
-		return cipher.NewCBCEncrypter(block, iv), nil
+	if decrypt {
+		return cipher.NewCBCDecrypter(block, iv), nil
 	}
 
-	return cipher.NewCBCDecrypter(block, iv), nil
+	return cipher.NewCBCEncrypter(block, iv), nil
 }
 
 // Pad text to blockSize.
 func Pad(text []byte, originalTextSize, blockSize int) []byte {
-	// pad to block size if needed.
+	// pad to block size if needed. The value of missing is between 0 and blockSize.
 	missing := blockSize - (originalTextSize % blockSize)
-	if missing > 0 && originalTextSize > 0 {
+	if missing > 0 {
```

Comment:
@thaidn I've updated the padding to always pad, even text with size = blockSize will be padded with a full dummy block with byte value=16

See sample outputs below:
`plaintext to pad: []
  output: [16 16 16 16 16 16 16 16 16 16 16 16 16 16 16 16]`
`plaintext to pad: [0]
  output: [0 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15]`
`plaintext to pad: [0 1]
  output: [0 1 14 14 14 14 14 14 14 14 14 14 14 14 14 14]`
`plaintext to pad: [0 1 2]
  output: [0 1 2 13 13 13 13 13 13 13 13 13 13 13 13 13]`
`plaintext to pad: [0 1 2 3]
  output: [0 1 2 3 12 12 12 12 12 12 12 12 12 12 12 12]`
 ...
`plaintext to pad: [0 1 2 3 4 5 6 7 8 9 10 11 12 13 14]
 output: [0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 1]`
`plaintext to pad: [0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15]
 output: [0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 16 16 16 16 16 16 16 16 16 16 16 16 16 16 16]`
`plaintext to pad: [0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16]
 output: [0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15]`
 ...
`plaintext to pad: [0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30]
 output: [0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 1]`
`plaintext to pad: [0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31]
 output: [0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 16 16 16 16 16 16 16 16 16 16 16 16 16 16 16 16]`
`plaintext to pad: [0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32]
output: [0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 15 15 15 15 15 15 15 15 15 15 15 15 15 15 15]`
 ...